core/src/com/ericc/the/game/map/LevelFactory.java

```diff
@@ -1,46 +1,45 @@
 package com.ericc.the.game.map;
 
 import com.badlogic.ashley.core.Entity;
-import com.ericc.the.game.Media;
 import com.ericc.the.game.entities.Mob;
 import com.ericc.the.game.entities.PushableObject;
 import com.ericc.the.game.entities.Stairs;
 
 import java.util.ArrayList;
 
 public class LevelFactory {
-    private final static int ROOM_WIDTH = 30;
-    private final static int ROOM_HEIGHT = 30;
-    private final static int ROOM_SIZE = 9;
-    private final static int MOBS_COUNT = 10;
+    private final static int ROOM_WIDTH = 40;
+    private final static int ROOM_HEIGHT = 40;
+    private final static int ROOM_SIZE = 20;
+    private final static int MOBS_COUNT = 20;
     private final static int CRATES_COUNT = 10;
 
     public static Level generate(int levelNumber) {
         Map map = new MapGenerator(ROOM_WIDTH, ROOM_HEIGHT, ROOM_SIZE).generateMap();
 
         ArrayList<Entity> entities = new ArrayList<>();
         for (int i = 0; i < MOBS_COUNT; i++) {
-            entities.add(new Mob(map.getRandomPassableTile()));
+            entities.add(new Mob(map.getRandomPassableTile(), map));
         }
 
         for (int i = 0; i < CRATES_COUNT; i++) {
-            entities.add(new PushableObject(map.getRandomPassableTile(), Media.crate));
+            entities.add(new PushableObject(map.getRandomPassableTile(), map));
         }
 
-        Stairs exit = new Stairs(map.getRandomPassableTileFromRooms(),
+        Stairs exit = new Stairs(map.getRandomPassableTileFromRooms(), map,
                 StaircaseDestination.DESCENDING
         );
 
         entities.add(exit);
-        map.registerStairs(exit);
+        map.registerStairs(exit.pos, StaircaseDestination.DESCENDING);
```

Comment:
Po to to wczesniej było tak zrobione, żeby nie przekazywać dwa razy tego samego argumentu czyt. DESCENDING (wyzej i nizej, to samo z ASCENDING) -> revert do poprzedniego

Reply:
Skrócenie tej linijki o jeden argument nie jest warte wprowadzania nowego komponentu.

Reply:
Ja dorzucę od siebie, że to bez sensu i brzydko żeby ta funkcja brała PositionComponent. Niech bierze GridPointa2d

Reply:
Natomiast w szerszej perspektywie to pokazuje nasz problem z nietrzymaniem się jednej spójnej wersji w sprawie GridPointa2D: PositionComponent nie używa GridPointa, tylko x i y.

Reply:
Dlatego wydaje mi się, że powinniśmy wywalić pary x i y zewsząd i wszędzie używać GridPointa2D.

Reply:
Jedynym efektem tego, że wymienimy x i y na GridPoint2D bedzie to, że zamiast `pos.x`, `pos.y` będziemy musieli pisać `pos.coś.x`, `pos.coś.y`.
Proponowałbym raczej wywalić zewsząd GridPoint2D, bo nie ma z niego żadnego pożytku, a dokłada konstruktorów.